WordPress/src/main/java/org/wordpress/android/ui/reader/usecases/ReaderCommentsFollowUseCase.kt

```diff
@@ -0,0 +1,124 @@
+                    postSubscribersApiCallsProvider.getMySubscriptionToPost(blogId, postId, continuation)
+                }
+
+                when (status) {
+                    is Success -> {
+                        emit(
+                                FollowCommentsState.FollowStateChanged(
+                                        blogId,
+                                        postId,
+                                        status.isFollowing,
+                                        isInit
+                                )
+                        )
+                    }
+                    is Failure -> {
+                        emit(FollowCommentsState.Failure(blogId, postId, true, UiStringText(status.error)))
+                    }
+                }
+            }
+        }
+    }
+
+    suspend fun setMySubscriptionToPost(
+        blogId: Long,
+        postId: Long,
+        subscribe: Boolean
+    ): Flow<FollowCommentsState> = flow {
+        emit(FollowCommentsState.Loading)
+
+        if (!networkUtilsWrapper.isNetworkAvailable()) {
+            emit(FollowCommentsState.NoNetwork(false, UiStringRes(R.string.error_network_connection)))
+        } else {
+            val status: PostSubscribersCallResult = suspendCoroutine { continuation ->
+                if (subscribe) {
+                    postSubscribersApiCallsProvider.subscribeMeToPost(blogId, postId, continuation)
+                } else {
+                    postSubscribersApiCallsProvider.unsubscribeMeFromPost(blogId, postId, continuation)
+                }
+            }
+
+            when (status) {
+                is Success -> {
+                    emit(
+                            FollowCommentsState.FollowStateChanged(
+                                    blogId,
+                                    postId,
+                                    status.isFollowing,
+                                    false,
+                                    UiStringRes(
+                                        if (status.isFollowing)
+                                            R.string.reader_follow_comments_subscribe_success
+                                        else
+                                            R.string.reader_follow_comments_unsubscribe_success
+                                    )
+                            )
+                    )
+                }
+                is Failure -> {
+                    emit(FollowCommentsState.Failure(blogId, postId, false, UiStringText(status.error)))
+                }
+            }
+        }
+    }
+
+    sealed class FollowCommentsState {
+        object Loading : FollowCommentsState()
+
+        data class FollowStateChanged(
+            val blogId: Long,
+            val postId: Long,
+            val isFollowing: Boolean,
+            val isInit: Boolean = false,
+            val userMessage: UiString? = null
+        ) : FollowCommentsState()
+
+        data class NoNetwork(
```

Comment:
I think I'd merge this and the `Failure` class. They are handled very similarly. It's just a type of a Failure

Reply:
Good point 👍 . Merged it in 015dd2a8c791d38eb3427bb80fbcb04d8ef3d329 . Also used `UpdateFollowCommentsUiState` as a simple data class removing the Nop state.